src/makeCalculator.js

```diff
@@ -37,7 +37,64 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  function add(num) {
+    result += num;
+  }
+
+  function subtract(num) {
+    result -= num;
+  }
+
+  function multiply(num) {
+    result *= num;
+  }
+
+  function divide(num) {
+    result /= num;
+  }
+
+  function reset() {
+    result = 0;
+
+    return this; // дозволяємо використовувати методи в ланцюжку
+  }
+
+  function operate(callback, num) {
+    callback(num);
+
+    return this; // дозволяємо використовувати методи в ланцюжку
+  }
+
+  return {
+    add,
+    subtract,
+    multiply,
+    divide,
+    reset,
+    operate,
+    get result() {
+      return result;
+    },
+  };
```

Comment:
It is better to initially create an object with all these methods and return it here